pandas/io/pytables.py

```diff
@@ -1924,6 +1926,13 @@ def get_atom_coltype(self, kind=None):
         else:
             col_name = "%sCol" % kind.capitalize()
 
+        if kind.startswith('complex'):
```

Comment:
I suggest moving it up a level to ``set_atom`` (as you know that ``block.is_complex``) already